axlearn/common/launch.py

```diff
@@ -13,6 +13,11 @@
     "--xla_tpu_spmd_rng_bit_generator_unsafe=1",  # SPMD partition-aware RngBitGenerator.
     "--xla_tpu_enable_latency_hiding_scheduler=true",  # Try to schedule ops efficiently.
     "--xla_tpu_perform_spmd_cse_prevention=false",  # b/229655601: prevent OOM on gpt2-small-repeat.
+    # If MegaScale Runtime Error is encountered when running multi-slice jobs,
+    # enabling this flag will allow for termination of the job, triggering
+    # auto-restart of the nodes. This is set to true to prevent the job from
+    # silently hanging and to reduce time to recovery.
```

Comment:
What's the granularity of the termination? Is it restarting the node encountering the error or all nodes in the slice or job?

Reply:
Being that this flag triggers the process to exit (see other comment), with the current JobSet configuration, this would trigger a JobSet restart, which is implemented as a full recreation of all Pods.

Reply:
Pods would land on the same Nodes upon restart barring manual intervention.